src/app/state/STATE.MD

```diff
@@ -0,0 +1,86 @@
+
+### Organise data slices around related data
+
+Try to group data into slices of data that are related. Bigger data slices are preferable to lots of small slices, so first see if there is an existing slice where data can be added before starting a new slice. Having fewer slices helps to prevent duplication of data.
+
+At the same time, if a slice becomes too big and cumbersome, it might make sense to split out some subsection of the state into its own slice.
+
+### Save all files related to a slice in a seperate folder
+
+For each slice a seperate folder <slice_name> must be created in the 'state" folder. E.g. for the 'user' slice, a folder called .../state/user must be created. All files realted to that slice must be saved in this folder (see next principle).
+
+### Follow a consistent state file naming convention
+
+For each state slice, there should be a main file named `<slice_name>Slice.ts` e.g. userSlice.ts This file will contain the state definition, initial slice state, slice reducers, slice actions and slice selectors.
+
+If the slice has Thunks, these functions must be in a sepeate file called `<slice_name>Thunks.ts` Thunks can be either synchronous or asynchronous.
+
+If the slice has any functions that are required to make code in the previous 2 files more readable or more succint, then these functions must be placed in a separate file called `<slice_name>Utils.ts` The idea is to make the ...Slice and ...Thunks files as short and easy to read as possible, with long and complicated functions grouped in the ...Utils file. If it is required to have more than one ...Utils files for a slice, then keep the format of `<slice_name><description><Utils>.ts` for any extra files.
+
+### Follow a consistent file structure for ...Slice files
+
+The ...Slice file is the main file for any state slice and should provide an easy to understand interface of what fields and actions are available for a slice. To help facilitate this every ...Slice file must follow the following structure:
+
+- Imports
+- Declaration of any types and interfaces used in the slice
+- An interface for the state stored in the slice named `<slice_name>State`
+  ```
+  export interface userState = {
+    ...
+  }
+  ```
+- The initial state of the slice stored in a constant called `initialState`
+  ```
+  const initialState: userState = {
+    ...
+  }
+  ```
+- The createSlice function for the slice called `<slice_name>Slice`
+  ```
+  export const unsrSlice = createSlice({
+    name: "user",
+    initialState,
+    reducers: {
+      ...
+    },
+    extraReducers: (builder) => {
+      ...
+    }
+  })
+  ```
+- an export of the available actions for the slice called ```<slice_name>Actions. The purpose of this export is to only expose the actions you want users to call (which might exclude certain reducers and/or thunks that should only be used by the slice internally).
```

Comment:
Since this example slice is `userSlice` the actions are userActions. But in general this could be called `rewardActions` or `radixActions` and does not neccessarily represent user actions in all cases.

```suggestion
- an export of the available actions for the slice called ```<slice_name>Actions. The purpose of this export is to list all actions that can be called from outside of the slice (which might exclude certain reducers and/or thunks that should only be used by the slice internally).
```